lib/gitsh/tab_completion/README.md

```diff
@@ -102,6 +102,21 @@ then hit <kbd>tab</kbd>. In this case, we'd pass `add` to the automaton and move
 from state `0` to state `4`. From state `4` the next thing we expect is a path,
 so we'd offer file paths from the current directory as completions.
 
+### Building the state graph
+
+The state graph is built by the `Gitsh::TabCompletion::AutomatonFactory` class.
+The initial implementation is small and fixed, but is expected to grow in size
+and complexity as we support more subtle forms of tab completion.
```

Comment:
I'm not excited about this sentence/I worry about it staying around longer than this graph's simplicity.